code/datums/components/aura_healing.dm

```diff
@@ -0,0 +1,168 @@
+#define HEAL_EFFECT_COOLDOWN (1 SECONDS)
+
+/// Applies healing to those in the area.
+/// Will provide them with an alert while they're in range, as well as
+/// give them a healing particle.
+/// Can be applied to those only with a trait conditionally.
+/atom/movable/screen/alert/aura_healing
+	name = "Aura Healing"
+	icon_state = "template"
+
+/datum/component/aura_healing
+	/// The range of which to heal
+	var/range
+
+	/// Whether or not you must be a visible object of the parent
+	var/requires_visibility = TRUE
+
+	/// Brute damage to heal over a second
+	var/brute_heal = 0
+
+	/// Burn damage to heal over a second
+	var/burn_heal = 0
+
+	/// Toxin damage to heal over a second
+	var/toxin_heal = 0
+
+	/// Suffocation damage to heal over a second
+	var/suffocation_heal = 0
+
+	/// Stamina damage to heal over a second
+	var/stamina_heal = 0
+
+	/// Amount of cloning damage to heal over a second
+	var/clone_heal = 0
+
+	/// Amount of blood to heal over a second
+	var/blood_heal = 0
+
+	/// Map of organ (such as ORGAN_SLOT_BRAIN) to damage heal over a second
+	var/list/organ_healing = null
+
+	/// Shows that aura heals fractures
+	var/heal_fractures = FALSE
+
+	/// Shows that aura heals internal bleedings
+	var/heal_internal_bleedings = FALSE
+
+	/// Showa that aura heals critical burns
```

Comment:
I don't think the wording is correct, it should probably be something like `Should this aura heal [x]`